code/__HELPERS/lists.dm

```diff
@@ -57,11 +57,17 @@
 	return 0
 
 //Checks for specific types in a list
-/proc/is_type_in_list(var/atom/A, var/list/L)
+/proc/is_type_in_list(atom/A, list/L)
+	if (!L.len)
+		return 0
+	if (!isnum(L[L[1]])) //Has this not been converted to an associative list?
+		generate_type_list_cache(L) //Convert it to an associative list
+	return L[A.type]
```

Comment:
What happens when you pass in null, because I don't think this preserves behaviour currently.